dm-manager/pkg/grpc/grpc_server/grpc_server.go

```diff
@@ -0,0 +1,260 @@
+		InventoryClientService: InventoryClientService{
+			invClient: *invClient,
+		},
+		rbac:        rbacPolicy,
+		authEnabled: enableAuth,
+	}, nil
+}
+
+func (dms *DeviceManagementService) ReportAMTStatus(ctx context.Context, req *pb.AMTStatusRequest) (*pb.AMTStatusResponse, error) {
+
+	zlog.Debug().Msgf("ReportAMTStatus started")
+
+	if dms.authEnabled {
+		if !dms.rbac.IsRequestAuthorized(ctx, rbac.CreateKey) {
+			err := inv_errors.Errorfc(codes.PermissionDenied, "Request is blocked by RBAC")
+			zlog.InfraSec().InfraErr(err).Msgf("Request Device management is not authenticated")
+		}
+	}
+
+	tenantID, present := inv_tenant.GetTenantIDFromContext(ctx)
+	if !present {
+		err := inv_errors.Errorfc(codes.Unauthenticated, "Tenant ID is missing from context")
+		zlog.InfraSec().InfraErr(err).Msgf("Request Nodes is not authenticated")
+		return nil, err
+	}
+	zlog.Debug().Msgf("ReportAMTStatus: tenantID=%s", tenantID)
+
+	hostInv, err := dms.invClient.GetHostByUUID(ctx, tenantID, req.HostId)
+	if err != nil {
+		zlog.InfraSec().InfraErr(err).Msgf("Failed to get host by UUID %s", req.HostId)
+		if inv_errors.IsNotFound(err) {
+			zlog.Debug().Msgf("Node Doesn't Exist for UUID %s and tID=%s\n",
+				req.HostId, tenantID)
+			return nil, inv_errors.Errorfc(codes.NotFound, "Host with UUID %s not found", req.HostId)
+		}
+	}
+	zlog.Debug().Msgf("Request from PMA=%s", req.GetStatus().String())
+	zlog.Debug().Msgf("hostInv AMTStatus=%s", hostInv.AmtStatus)
+	if hostInv.AmtStatus != req.GetStatus().String() {
+		err = dms.updateHost(ctx, hostInv.GetTenantId(), hostInv.GetResourceId(),
+			&fieldmaskpb.FieldMask{Paths: []string{
+				computev1.HostResourceFieldAmtStatus,
+			}}, &computev1.HostResource{
+				AmtStatus: req.GetStatus().String(),
+			})
+		if err != nil {
+			zlog.InfraSec().InfraErr(err).Msgf("Failed to update AMT status for host %s", hostInv.GetResourceId())
+			return nil, inv_errors.Errorfc(codes.Internal, "Failed to update AMT status: %v", err)
+		}
+		return &pb.AMTStatusResponse{}, nil
+	}
+
+	return nil, inv_errors.Errorfc(codes.FailedPrecondition, "AMT status is already set to %s", hostInv.AmtStatus)
+}
+
+func (dms *DeviceManagementService) RetrieveActivationDetails(ctx context.Context, req *pb.ActivationRequest) (*pb.ActivationDetailsResponse, error) {
+
+	zlog.Info().Msgf("RetrieveActivationDetails")
+
+	if dms.authEnabled {
+		if !dms.rbac.IsRequestAuthorized(ctx, rbac.CreateKey) {
+			err := inv_errors.Errorfc(codes.PermissionDenied, "Request is blocked by RBAC")
+			zlog.InfraSec().InfraErr(err).Msgf("Request Device management is not authenticated")
+		}
+	}
+
+	tenantID, present := inv_tenant.GetTenantIDFromContext(ctx)
+	if !present {
+		err := inv_errors.Errorfc(codes.Unauthenticated, "Tenant ID is missing from context")
+		zlog.InfraSec().InfraErr(err).Msgf("Request Nodes is not authenticated")
+		return nil, err
+	}
+	zlog.Debug().Msgf("ReportAMTStatus: tenantID=%s", tenantID)
+
+	var hostInv *computev1.HostResource
```

Comment:
This variable setting is not needed, hostInv can be directly set from the response from `dms.invClient.GetHostByUUID`